packages/@uppy/xhr-upload/src/index.js

```diff
@@ -180,8 +180,13 @@ module.exports = class XHRUpload extends Plugin {
       formPost.append(item, file.meta[item])
     })
 
+    // Set `file.data.type` in the blob to `file.meta.type`,
+    // because we might have detected a more accurate file type in Uppy
+    // https://stackoverflow.com/a/50875615
+    file.data = file.data.slice(0, file.data.size, file.meta.type)
```

Comment:
maybe we should have a reused `copyWithNewType` function? So that the comment and stackoverflow reference will only live in one place? And also the name is friendlier to read throughout the code

Reply:
Good idea, refactored, thanks!